cmd/tools/genvisitor/main.go

```diff
@@ -43,10 +43,18 @@ func main() {
 			}
 			// Skip nested "Cause" field in Failure types. The repairInvalidUTF8InFailure handler function
 			// will descend into these.
-			if strings.Contains(path.String(), "/Cause") {
+			ps := path.String()
+			if strings.Contains(ps, "/Cause") {
 				logger.Debug("ignore failure Cause", tag.NewAnyTag("path", path.String()))
 				return false
 			}
+
+			// These do not have a failure field in Temporal v1.22 (they do in later versions)
+			if strings.Contains(ps, "WorkflowQueryResult") ||
+				strings.Contains(ps, "RespondQueryTaskCompletedRequest") ||
```

Comment:
so `WorkflowQueryResult` and `RespondQueryTaskCompletedRequest` are in Failure type? how did you discovered? (to help future version upgrade)

should we just move all these fields into an array and filter that way?